src/services/codefixes/importFixes.ts

```diff
@@ -371,37 +453,89 @@ namespace ts.codefix {
         }
     }
 
-    function tryAddToExistingImport(existingImports: readonly FixAddToExistingImportInfo[], canUseTypeOnlyImport: boolean): FixAddToExistingImport | undefined {
-        return firstDefined(existingImports, ({ declaration, importKind }): FixAddToExistingImport | undefined => {
-            if (declaration.kind === SyntaxKind.ImportEqualsDeclaration) return undefined;
+    function getAddAsTypeOnly(
+        isValidTypeOnlyUseSite: boolean,
+        isForNewImportDeclaration: boolean,
+        symbol: Symbol,
+        targetFlags: SymbolFlags,
+        checker: TypeChecker,
+        compilerOptions: CompilerOptions
+    ) {
+        if (!isValidTypeOnlyUseSite) {
+            // Can't use a type-only import if the usage is an emitting position
+            return AddAsTypeOnly.NotAllowed;
+        }
+        if (isValidTypeOnlyUseSite && isForNewImportDeclaration && compilerOptions.importsNotUsedAsValues === ImportsNotUsedAsValues.Error) {

```

Comment:
`isValidTypeOnlyUseSite` can only be true here. I guess you might still want to mention it even if it's a redundant check.

Reply:
It was probably overlooked as part of a refactor, I’m just as happy deleting it.